### PR TITLE
Fix updateOrCreate

### PR DESCRIPTION
**Pull request by ssh24**
### Description

Allow use of `updateAttributes` on the `updateOrCreate` prototype function. Try updating the data using the `updateAttributes` function first and if it fails, create the instance.

connect to https://github.com/strongloop-internal/scrum-apex/issues/204

### CODE REVIEW

File: lib/cloudant.js [hunk 1]
```diff
@@ -710,7 +710,7 @@ Cloudant.prototype.updateAttributes = function(model, id, data, options, cb) {
     if (err) return cb(err);
     data = getPlainJSONData.call(self, model, data);
     _.mergeWith(doc, data, function(dest, src) { return src; });
-    self.create(model, data, options, function(err) {
+    self.create(model, doc, options, function(err) {
```
Comment (b-admike) [inline lib/cloudant.js]: Good catch!

File: test/cloudant.test.js [hunk 2]
```diff
@@ -117,6 +117,64 @@ describe('cloudant connector', function() {
...
+        should.not.exist(err);
+        should.exist(result);
+        result.id.should.equal(newData.id);
+        result.name.should.equal(newData.name);
+        result.age.should.equal(newData.age);
+        done();
+      });
+    });
+
+    it('update existing instance if instance exists', function(done) {
```
Comment (b-admike) [inline test/cloudant.test.js]: I would update one of the instances created by the `create` function in L#136

Comment (ssh24) [inline test/cloudant.test.js]: I agree. Fixed it.

**Comment by dhmlau** [comment]
Not sure how it's done in other connectors.. For updateOrCreate, would the logic be the following instead: 
- check if the instance exists 
  - if exists, update
  - if not exist, create

**Comment by ssh24** [comment]
@dhmlau That is exactly what we did on this PR. `updateAttributes` throws an error **404** if the instance does not exist, so we check that before creation and if it does not throw an error, the instance has been updated.

**Comment by dhmlau** [comment]
I was thinking if there’s a better function to check the existence of the instance instead of calling the `updateAttributes` to find out

**Comment by ssh24** [comment]
I am not sure why we would want to do another call to a function to check the existence and then call `updateAttributes` later on when calling `updateAttributes` first already does half the job for us. We can catch the error in `updateAttributes` and if it returns a non-existent code we create the instance.

**Comment by dhmlau** [comment]
that's true too.  I was thinking what if the `updateAttribute` function failed for a different reason, the error message we're displaying to the users might be different?  
Anyway, I'm ok either way.  Thanks.

**Pull request merged**